### PR TITLE
Add types for chrome.i18n

### PR DESCRIPTION
**Pull request by ssorallen**
Add unimplemented types for [`chrome.i18n`][0]. Types are all based off
Chrome's documentation.

Addresses #1

[0]: https://developer.chrome.com/extensions/i18n

### CODE REVIEW

File: interfaces/i18n.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+type chrome$LanguageCode = string;
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (Array<chrome$LanguageCode>) => void): void,
```
Comment (Shraymonks) [inline interfaces/i18n.js]: Name the callback parameter?
```javascript
getAcceptLanguages(callback: (languages: Array<chrome$LanguageCode>) => void): void,
```

File: interfaces/i18n.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  getMessage(messageName: string, ...substitutions?: Array<*>): ?string,
```
Comment (Shraymonks) [inline interfaces/i18n.js]: `substitutions` is just an array parameter, not rest parameters https://developer.chrome.com/extensions/i18n#method-getMessage

File: interfaces/i18n.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  getUILanguage(): string,
+  detectLanguage(text: string, callback: (result: {
+    isReliable: boolean,
+    languages: {
```
Comment (Shraymonks) [inline interfaces/i18n.js]: `languages` is an array of these objects
https://developer.chrome.com/extensions/i18n#method-detectLanguage

**Review by Shraymonks — CHANGES_REQUESTED** [review]
Thanks for your work on this!

File: interfaces/i18n.js (hunk 2 above)
Comment (ssorallen) [reply interfaces/i18n.js]: Nice catch, thanks.

I'm using the API, and I missed the note that it wasn't rest params. It actually works with a single string, but more than one requires an Array. The examples even mention the single string case:

https://developer.chrome.com/apps/i18n#examples-getMessage

It seems best to document the example and the actual API even though their documentation doesn't agree.

What do you think?

**Comment by ssorallen** [comment]
Thanks for the feedback. I addressed each point.

File: interfaces/i18n.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+};
+
+type chrome$LanguageDetectionResult = {
+  isReliable: boolean,
+  languages: Array<chrome$DetectedLanguage>,
+};
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (languages: Array<chrome$LanguageCode>) => void): void,
+  getMessage: (messageName: string, substitutions?: string | Array<string>) => string,
```
Comment (Shraymonks) [inline interfaces/i18n.js]: `getMessage` can return `undefined`

Comment (Shraymonks) [inline interfaces/i18n.js]: Change this back to the method annotation style
```js
getMessage(...): type,
```

**Review by Shraymonks — CHANGES_REQUESTED** [review]

File: interfaces/i18n.js (hunk 4 above)
Comment (ssorallen) [reply interfaces/i18n.js]: Oh yes thanks. I mistakenly changed that.

Comment (ssorallen) [reply interfaces/i18n.js]: Will do.

**Comment by ssorallen** [comment]
@Shraymonks All set.

**Review by Shraymonks — APPROVED** [review]

**Comment by Shraymonks** [comment]
@ssorallen Thanks!

**Pull request merged**